### ISSUE TITLE
On linux not able to create multiple receivers

### ISSUE THREAD

**Issue by ahtpatel** [Azure/azure-event-hubs-python#64]
While running recv_async.py on linux I get the below error messages and I am able to receive only from a single partition. The same code when run on windows works without any issues.

(eventhub) [root@aht-S10 aht]# python -V
Python 3.6.3
(eventhub) [root@aht-S10 aht]# pip freeze
adal==1.1.0
asn1crypto==0.24.0
azure-common==1.1.15
azure-eventhub==1.0.0
azure-nspkg==2.0.0
azure-storage==0.36.0
certifi==2018.8.24
cffi==1.11.5
chardet==3.0.4
cryptography==2.3.1
entrypoints==0.2.3
idna==2.7
isodate==0.6.0
jeepney==0.3.1
keyring==15.1.0
msrest==0.5.5
msrestazure==0.4.34
oauthlib==2.1.0
pycparser==2.19
PyJWT==1.6.4
python-dateutil==2.7.3
requests==2.19.1
requests-oauthlib==1.0.0
SecretStorage==3.1.0
six==1.11.0
uamqp==1.0.3
urllib3==1.23
virtualenv==15.1.0

[log.txt](https://github.com/Azure/azure-event-hubs-python/files/2397701/log.txt)

Error messages:

2018-09-19 15:05:29,465 azure.eventhub.async_ops INFO     Encountered error while starting handler: ValueError('The supplied authentication has already been consumed byanother connection. Please create a fresh instance.',)
..
..
2018-09-19 15:05:30,929 azure.eventhub.async_ops WARNING  'eventhub.pysdk-bf3b8e00': 3 clients failed to start.

recv_async.py:

```
ADDRESS = "xxx"
USER = "xxx"
KEY = "xxx"
CONSUMER_GROUP = "$default"
OFFSET = Offset("-1")
lis = []


async def pump(client, partition):
    receiver = client.add_async_receiver(CONSUMER_GROUP, partition, OFFSET, prefetch=200)
    await client.run_async()
    total = 0
    start_time = time.time()
    while True:

        batch = await receiver.receive(max_batch_size=100, timeout=10)
        for event_data in batch:
            msg = json.loads(str(event_data.message))
            print(msg)

try:
    if not ADDRESS:
        raise ValueError("No EventHubs URL supplied.")

    loop = asyncio.get_event_loop()
    client = EventHubClientAsync(ADDRESS, debug=False, username=USER, password=KEY)
    tasks = [
        asyncio.ensure_future(pump(client, "0")),
        asyncio.ensure_future(pump(client, "1")),
        asyncio.ensure_future(pump(client, "2")),
        asyncio.ensure_future(pump(client, "3"))]
    loop.run_until_complete(asyncio.wait(tasks))
    loop.run_until_complete(client.stop_async())
    loop.close()
```

**Comment by annatisch** [comment]
Thanks @ahtpatel!

I think the error might be occurring in the multiple calling of client.run_async... I will need to dig into that code.
In the meantime, do you get the same error if you configure the pumps beforehand? Something like this:
```
async def pump(receiver):
    total = 0
    start_time = time.time()
    while True:
        batch = await receiver.receive(max_batch_size=100, timeout=10)
        for event_data in batch:
            msg = json.loads(str(event_data.message))
            print(msg)

try:
    if not ADDRESS:
        raise ValueError("No EventHubs URL supplied.")

    loop = asyncio.get_event_loop()
    client = EventHubClientAsync(ADDRESS, debug=False, username=USER, password=KEY)
    tasks = []
    for i in range(4):
        receiver = client.add_async_receiver(CONSUMER_GROUP, str(i), OFFSET, prefetch=200)
        tasks.append(asyncio.ensure_future(pump(receiver))
    loop.run_until_complete(client.run_async())
    loop.run_until_complete(asyncio.wait(tasks))
    loop.run_until_complete(client.stop_async())
    loop.close()
```

**Comment by ahtpatel** [comment]
Hi @annatisch ,
The above code does not work. I still get the same error. The weird part is when I run the same code on a windows box it works perfectly.

**Comment by annatisch** [comment]
Thanks @ahtpatel - I will investigate today.
Currently Windows and Linux are using two different TLS layers so that's a good indication that this must be where the issue is.

**Comment by annatisch** [comment]
I have found and fixed the issue (I'm actually surprised we're not seeing it in Windows as well - maybe there's differences in the handling of async functions between the two OSs).

I will publish a patch this week.
Thanks for the report!

**Comment by ahtpatel** [comment]
Hi @annatisch ,

I tried it again with the code given by you previously. I still get the error. Please find script command output attached.
[log.txt](https://github.com/Azure/azure-event-hubs-python/files/2405079/log.txt)

**Comment by annatisch** [comment]
Hi @ahtpatel - I think I have found the issue - in this line here:
```
tasks.append(asyncio.ensure_future(pump(receiver)))
```

By calling `asyncio.ensure_future` here, it's actually scheduling the pump function on the event loop *before* the client has been started. This will attempt to start up the underlying receiver handlers anyway - then when `client.run_async` is called it attempts to start them again.

If you replace the line with just:
```
tasks.append(pump(receiver)))
```

This should resolve it :)

I will make a couple of amendments to the receiver code to present a useful error if the receiver tries to receiver before the client has been started.

**Comment by annatisch** [comment]
Hi @ahtpatel - I hope the suggestion is my last post helped!
I've also now published v1.1.0 that wont attempt to start the same handler twice which should avoid the original error :)

Thanks for the report!

**Comment by ahtpatel** [comment]
Hi @annatisch,
Yep, it worked. Thanks
